core/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -1411,19 +1431,33 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
         DeltaSQLConf.DELTA_PROTOCOL_DEFAULT_WRITER_VERSION.key -> "1") {
         spark.range(10).writeTo(s"delta.`${dir.getCanonicalPath}`")
           .tableProperty("delta.appendOnly", "true")
-          .tableProperty("delta.enableChangeDataFeed", "true")
           .using("delta")
           .create()
-        assert(deltaLog.update().protocol === Protocol(1, 4))
+        val protocolOfNewTable = Protocol(1, 2)
+        assert(deltaLog.update().protocol === protocolOfNewTable)
 
         val e = intercept[DeltaTableFeatureException] {
-          sql(
-            s"ALTER TABLE delta.`${dir.getCanonicalPath}` SET TBLPROPERTIES (" +
-              s"  '${TestReaderWriterMetadataAutoUpdateFeature.TABLE_PROP_KEY}' = 'true'," +
-              s"  '${TestWriterMetadataNoAutoUpdateFeature.TABLE_PROP_KEY}' = 'true')")
+          // ALTER TABLE must not consider this SQL config
+          withSQLConf(defaultPropertyKey(TestWriterFeature) -> "supported") {
+            sql(
+              s"ALTER TABLE delta.`${dir.getCanonicalPath}` SET TBLPROPERTIES (" +
+                s"  'delta.appendOnly' = 'false'," +
+                s"  'delta.enableChangeDataFeed' = 'true'," +
+                s"  '${TestReaderWriterMetadataAutoUpdateFeature.TABLE_PROP_KEY}' = 'true'," +
+                s"  '${TestWriterMetadataNoAutoUpdateFeature.TABLE_PROP_KEY}' = 'true')")
+          }
         }
 
+        val unsupportedFeatures = TestWriterMetadataNoAutoUpdateFeature.name
+        val supportedFeatures =
+          (protocolOfNewTable.implicitlyAndExplicitlySupportedFeatures +
+            ChangeDataFeedTableFeature +
+            TestReaderWriterMetadataAutoUpdateFeature).map(_.name).toSeq.sorted.mkString(", ")
         assert(e.getErrorClass === "DELTA_FEATURES_REQUIRE_MANUAL_ENABLEMENT")
+
+        assert(e.getMessage.contains(s" $unsupportedFeatures."))
```

Comment:
Can we test the parameters embedded in the DeltaTableFeatureException directly, instead of searching in the message? That's the clean way to do this.

Reply:
Getting parameters requires Spark 3.4, unfortunately :(

Reply:
Left a comment in the code.